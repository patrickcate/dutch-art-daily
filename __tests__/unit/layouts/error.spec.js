import Error from '@layouts/error.vue'

describe('Error', () => {
  let wrapper
  let options

  beforeEach(() => {
    options = {
      propsData: {
        error: {
          statusCode: '404',
        },
      },
    }

    wrapper = createWrapper(Error, options)
  })

  it('should be a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders generic message', () => {
    wrapper.setProps({
      error: null,
    })
    expect(wrapper).toMatchSnapshot()
  })
})
