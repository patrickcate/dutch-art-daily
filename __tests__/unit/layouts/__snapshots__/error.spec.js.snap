// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error does not render image if src path is not available 1`] = `
<div class="error">
  <h1 class="error__header">Oops, an error occurred</h1>
  <h2 class="error__subheader">Maybe someone was sleeping on the job?</h2>
  <!---->
  <p class="error__message">
    You'll have better luck
    <a href="/" rel="home">going to the Homepage</a></p>
</div>
`;

exports[`Error renders correctly 1`] = `
<div class="error">
  <h1 class="error__header">Oops, an error occurred</h1>
  <h2 class="error__subheader">Maybe someone was sleeping on the job?</h2>
  <div class="artwork-image-frame artwork--portrait"><img alt="Nicolaes Maes's painting: The Idle Servant" src="/photos/01-02/01-02--xs3-8BAoia4yi0O.jpg" class="artwork-image error__image"></div>
  <p class="error__message">
    You'll have better luck
    <a href="/" rel="home">going to the Homepage</a></p>
</div>
`;

exports[`Error renders generic message 1`] = `
<div class="error">
  <h1 class="error__header">Oops, an error occurred</h1>
  <h2 class="error__subheader">Maybe someone was sleeping on the job?</h2>
  <div class="artwork-image-frame artwork--portrait"><img alt="Nicolaes Maes's painting: The Idle Servant" src="/photos/01-02/01-02--xs3-8BAoia4yi0O.jpg" class="artwork-image error__image"></div>
  <p class="error__message">
    You'll have better luck
    <a href="/" rel="home">going to the Homepage</a></p>
</div>
`;
